hibernate-core/src/main/java/org/hibernate/query/sqm/function/AbstractSqmFunctionDescriptor.java

```diff
@@ -54,6 +47,36 @@ public AbstractSqmFunctionDescriptor(ArgumentsValidator argumentsValidator, Func
 				: returnTypeResolver;
 	}
 
+	public String getSignature(String functionName) {
+		return getReturnSignature() + functionName + getArgumentListSignature();
+	}
+
+	/**
+	 * The return type of the function in a format suitable
+	 * for display to the user.
+	 */
+	public String getReturnSignature() {
+		String result = returnTypeResolver.getReturnType();
+		return result.isEmpty() ? "" : result + " ";
+	}
+
+	/**
+	 * The argument list of the function in a format suitable
+	 * for display to the user.
+	 */
+	public String getArgumentListSignature() {
+		String args = argumentsValidator.getSignature();
+		return requiresArgumentList() ? args : "()".equals(args) ? "" : "[" + args + "]";
```

Comment:
it seems adding a parentheses for the upper level sub expression would improve code readability:
```
return requiresArgumentList() ? args : ( "()".equals(args) ? "" : "[" + args + "]" );
```